src/Bicep.Core/FileSystem/IFileResolver.cs

```diff
@@ -28,15 +28,12 @@ public interface IFileResolver
     /// <param name="fileUri">The file URI to read.</param>
     ResultWithDiagnostic<string> TryRead(Uri fileUri);
 
+    ResultWithDiagnostic<string> TryReadAtMostNCharacters(Uri fileUri, Encoding fileEncoding, int n);
```

Comment:
[nit]: this one is missing a doc string which is inconsistent with the rest of them